core/typedefs.h

```diff
@@ -98,25 +98,30 @@
 #endif
 
 // Generic ABS function, for math uses please use Math::abs.
-#ifndef ABS
-#define ABS(m_v) (((m_v) < 0) ? (-(m_v)) : (m_v))
-#endif
+template <typename T>
+constexpr T ABS(T m_v) {
```

Comment:
iOS seems to be failing due to a name clash with a builtin `ABS` define:
```
 ./core/typedefs.h:102:13: error: expected unqualified-id [4]
 constexpr T ABS(T m_v) {
             ^
/Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS15.2.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSObjCRuntime.h:611:20: note: expanded from macro 'ABS'
     #define ABS(A) __NSABS_IMPL__(A,__COUNTER__)
                    ^
/Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS15.2.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSObjCRuntime.h:610:34: note: expanded from macro '__NSABS_IMPL__'
     #define __NSABS_IMPL__(A,L) ({ __typeof__(A) __NSX_PASTE__(__a,L) = (A); (__NSX_PASTE__(__a,L) < 0) ? - __NSX_PASTE__(__a,L) : __NSX_PASTE__(__a,L); })
                                  ^
```

We should likely `#undef` all 5 macros to make room for our constexpr's.

Reply:
Done.